src/packages/solid/components/Badge/Badge.styles.ts

```diff
@@ -17,26 +17,116 @@
 
 import theme from 'theme';
 import { type NodeStyles } from '@lightningjs/solid';
+import type { Tone } from 'types';
+import type { ComponentStyleConfig, NodeStyleSet, TextStyleSet } from '../../types/types.js';
+import { makeComponentStyles } from '../../utils/index.js';
 
-const styles = {
-  Container: {
+export interface BadgeStyle {
+  tone: Tone;
+  Container: NodeStyleSet;
+  Icon: NodeStyleSet;
+  Text: TextStyleSet;
+}
+
+type BadgeStyleProperties = Partial<{
+  backgroundColor: NodeStyles['color'];
+  textColor: NodeStyles['color'];
+  iconColor: NodeStyles['color'];
+  strokeColor: NodeStyles['color']; // TODO do we use this?
+  borderColor: NodeStyles['color'];
+}>;
+
+type BadgeConfig = ComponentStyleConfig<BadgeStyleProperties>;
+
+/* @ts-expect-error next-line themes are supplied by client applications so this setup is necessary */
+const { Badge: { styles: themeStyles, tone } = { styles: {}, tone: 'neutral' } } = theme?.componentConfig;
+
+const container: BadgeConfig = {
+  themeKeys: {
+    color: 'backgroundColor'
+  },
+  base: {
+    color: theme.color.fillInverseSecondary,
+    borderRadius: theme.radius.sm, // borderRadius must be applied _before_ border to prevent the node from breaking
+    border: {
+      color: theme.color.strokeInverse,
+      width: theme.stroke.sm
+    },
     display: 'flex',
-    justifyContent: 'spaceEvenly',
     height: theme.typography.tag1.lineHeight,
```

Comment:
do we want to add a width?

Reply:
nah i figured out the issue. this height actually needs to be removed though